sae.c

```diff
@@ -480,16 +478,23 @@ process_confirm (struct candidate *peer, struct ieee80211_mgmt_frame *frame, int
 
     if (memcmp(tmp, (frame->authenticate.u.var8 + sizeof(unsigned short)), SHA256_DIGEST_LENGTH)) {
         sae_debug(SAE_DEBUG_ERR, "confirm did not verify!\n");
-        BN_free(x);
-        BN_free(y);
-        EC_POINT_free(psum);
-        return ERR_BLACKLIST;
+        r = ERR_BLACKLIST;
+        goto out;
     }
 
-    BN_free(x);
-    BN_free(y);
-    EC_POINT_free(psum);
-    return NO_ERR;
+    r = NO_ERR;
+
+    out: if (x) {
+      BN_free(x);
+    }
+    if (y) {
+      BN_free(y);
+    }
+    if (psum) {
+      EC_POINT_free(psum);
+    }
+
```

Comment:
These, like most free() functions, already check their argument for null, so you can just do:
```
out:
    BN_free(x);
    BN_free(y);
    EC_POINT_free(psum);
```

Also my preference would be if the "out" label were in column 0 instead of being indented.